### PR TITLE
KOGITO-6996 Added new guide: "Working with parallelism"

### PR DESCRIPTION
**Pull request by hbelmiro**
**JIRA:** https://issues.redhat.com/browse/KOGITO-6996

This PR depends on: https://github.com/kiegroup/kogito-examples/pull/1338

- [x] You have read the [contributions doc](https://github.com/kiegroup/kogito-docs/blob/main/CONTRIBUTING.md)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [x] The nav.adoc file has a link to this guide in the proper category

### CODE REVIEW

File: serverlessworkflow/modules/ROOT/pages/core/working-with-parallelism.adoc [hunk 1]
```diff
@@ -1 +1,279 @@
...
+:keywords: kogito, workflow, quarkus, serverless, parallelism
+
+This document describes how to run parallel tasks in Serverless Workflow.
+The testing procedure in this document is based on the `serverless-workflow-examples/serverless-workflow-service-calls-quarkus` example application.
+You can access this example application in the link:{kogito_sw_examples_url}/serverless-workflow-parallel-execution[GitHub repository].
+
+[WARNING]
+====
+Kogito Serverless Workflow serialize parallel execution. Therefore, "parallel" does not mean simultaneous execution. It means there is no logical dependency between branches. An inactive branch can start/resume execution without waiting completion of the active branch if the latter suspends execution (for example, wait for an event reception).
+====
```
Comment (hbelmiro) [inline serverlessworkflow/modules/ROOT/pages/core/working-with-parallelism.adoc:14]: @krisv could you take a look at this, please?

**Review by fjtirado — APPROVED** [review]
@hbelmiro Nice usage of existing integration test
https://github.com/kiegroup/kogito-runtimes/blob/main/quarkus/extensions/kogito-quarkus-serverless-workflow-extension/kogito-quarkus-serverless-workflow-integration-test/src/main/resources/parallel.sw.json
@krisv see helbers comment (actually my writing ;)). We were struggling to find out a way to express that limitation without compromising our reputation ;)

File: serverlessworkflow/modules/ROOT/pages/core/working-with-parallelism.adoc [hunk 2]
```diff
@@ -1 +1,279 @@
...
+
+The example application is a workflow that concatenates 3 strings and returns the result.
+The concatenation is done in parallel, one branch per each of the 3 strings.
+
+.Prerequisites
+The examples used in this document assumes that you have the following prerequisites:
+
+include::../../pages/_common-content/getting-started-requirement.adoc[]
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/core/working-with-parallelism.adoc:22]: I think we might remove this prereqs session.

**Review by ricardozanini — APPROVED** [review]
Greate guide! Well-written, clear examples and workflows. Very well done! Congrats!

Can you please add the card in the home?

**Comment by hbelmiro** [comment]
> Can you please add the card in the home?

Sure. I'll do that.
I created [this JIRA](https://issues.redhat.com/browse/KOGITO-7630) to add it to CONTRIBUTING.md.

**Comment by hbelmiro** [comment]
@ricardozanini
I pushed the suggested changes.
Thanks for your review.

**Comment by hmanwani-rh** [comment]
Merging this PR to create a patch PR with doc improvements.

**Pull request merged**